Makefile

```diff
@@ -92,80 +54,151 @@ build:
-	--exclude=/.git/ \
-	--exclude=/.idea/ \
-	--exclude=/build/ \
-	--exclude=/js/node_modules/ \
-	--exclude=*.log
-ifdef CAN_SIGN
-	$(sign) --path "$(source_build_directory)"
-else
-	@echo $(sign_skip_msg)
-endif
-	tar -cvzf $(source_package_name).tar.gz -C $(source_build_directory)/../ $(app_name)
-
-# Builds the source package for the app store, ignores php and js tests
+# Build the source package for the app store, ignores php and js tests
 .PHONY: appstore
-appstore: build
-	rm -rf $(appstore_build_directory) $(appstore_artifact_directory)
-	mkdir -p $(appstore_build_directory) $(appstore_artifact_directory)
-	$(copy_command) --parents -r \
-	"appinfo" \
-	"controller" \
-	"http" \
-	"img" \
-	"l10n" \
-	"templates" \
-	"css/public" \
-	"js/public" \
-	"COPYING" \
-	"CHANGELOG.md" \
-	$(appstore_build_directory)
+appstore: ## Build the source package for the app store
+	rm -rf $(dist_dir)
+	mkdir -p $(dist_dir)/$(app_name)
+	cp -R $(all_src) $(dist_dir)/$(app_name)
+
 ifdef CAN_SIGN
-	mv $(configdir)/config.php $(configdir)/config-2.php
-	$(sign) --path="$(appstore_build_directory)"
-	mv $(configdir)/config-2.php $(configdir)/config.php
+	$(sign) --path="$(dist_dir)/$(app_name)"
 else
 	@echo $(sign_skip_msg)
 endif
-	tar -czf $(appstore_package_name).tar.gz -C $(appstore_build_directory)/../ $(app_name)
+	tar -czf $(dist_dir)/$(app_name).tar.gz -C $(dist_dir) $(app_name)
+	tar -cjf $(dist_dir)/$(app_name).tar.bz2 -C $(dist_dir) $(app_name)
+
+$(nodejs_deps): package.json yarn.lock
+	yarn install && touch $@
+
+$(KARMA): $(nodejs_deps)
 
+##------------
+## Tests
+##------------
+
+# Command for running all tests.
+.PHONY: test
+test: test-php-unit test-js
 
 # Command for running JS and PHP tests. Works for package.json files in the js/
 # and root directory. If phpunit is not installed systemwide, a copy is fetched
-# from the internet
-.PHONY: test
-test:
-	cd js && $(yarn) run test
-ifneq ("$(wildcard $(phpunit_oc10))","")
-	php $(phpunit_oc10) -c phpunit.xml --coverage-clover coverage.clover
-else
-	phpunit -c phpunit.xml --coverage-clover coverage.clover
-	# phpunit -c phpunit.integration.xml --coverage-clover build/php-unit.clover
-endif
+# # from the internet
+# .PHONY: test
+# test:
+# 	cd js && $(yarn) run test
+# ifneq ("$(wildcard $(phpunit_oc10))","")
+# 	php $(phpunit_oc10) -c phpunit.xml --coverage-clover coverage.clover
+# else
+# 	phpunit -c phpunit.xml --coverage-clover coverage.clover
+# 	# phpunit -c phpunit.integration.xml --coverage-clover build/php-unit.clover
+# endif
+
+.PHONY: test-php-codecheck
+test-php-codecheck:
+	# currently failes - as we use a private api
+	#	$(occ) app:check-code $(app_name) -c private
+	$(occ) app:check-code $(app_name) -c strong-comparison
+	$(occ) app:check-code $(app_name) -c deprecation
+
+.PHONY: test-php-style
+test-php-style: ## Run php-cs-fixer and check owncloud code-style
+test-php-style: vendor-bin/owncloud-codestyle/vendor vendor-bin/php_codesniffer/vendor
```

Comment:
There are no acceptance tests, so we should remove all mentions of `php_codesniffer`
otherwise the style check gives:
```
ERROR: The file "tests/acceptance" does not exist.

Run "phpcs --help" for usage information

Makefile:146: recipe for target 'test-php-style' failed
```